### PR TITLE
(doc): add compression documentation part of #402

### PR DESCRIPTION
**Pull request by oneslash**
Adding compression documentation, I had spent some time to find the information is needed. I hope this will help others to start committing documentation too.

There are no server docs, since my server is on `Go` so I haven't explored that part.

### CODE REVIEW

**Comment by thelinuxfoundation** [comment]
Thank you for your pull request. Before we can look at your contribution, we need to ensure all contributors are covered by a Contributor License Agreement.

After the following items are addressed, please respond with a new comment here, and the automated system will re-verify.

* One or more commits are not associated with a GitHub user. The email used in the Git commit must match a verified email address in a GitHub user profile. For more information, please see https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/

_Regards,
CLA GitHub bot_

File: Documentation/compression.md [hunk 1]
```diff
@@ -0,0 +1,31 @@
+# Compression
+
+## Client side
+The preferred method for configuring message compression on a client is to pass `options` when the client object is instantiated.
+
+The two options need to be passed:
+
+**grpc.default_compression_algorithm** (integer)
+
+The option keeps the value of a compression algorithm.
```
Comment (murgatroid99) [inline Documentation/compression.md]: This should include a description of the functionality of this option. Based on the comments in [this header file](https://github.com/grpc/grpc/blob/master/include/grpc/impl/codegen/compression_types.h), I think it controls the default compression algorithm the channel will use to send messages.

File: Documentation/compression.md [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+Possible values for this option are:
+- `0` - No compression
+- `1` - Compress with DEFLATE algorithm
+- `2` - Compress with GZIP algorithm
+- `3` - Stream compression with GZIP algorithm
+
+**grpc.default_compression_level** (integer)
+
+The option keeps the value for the level of compression.
```
Comment (murgatroid99) [inline Documentation/compression.md]: Similarly, this should also describe the functionality of this option. I think this one controls the set of compression algorithms the channel will accept when reading messages, but I'm less sure of this one.

File: Documentation/compression.md [hunk 3]
```diff
@@ -0,0 +1,31 @@
+# Compression
+
+## Client side
+The preferred method for configuring message compression on a client is to pass `options` when the client object is instantiated.
+
+The two options need to be passed:
```
Comment (murgatroid99) [inline Documentation/compression.md]: I suggest something like "These two options control compression behavior". You don't need to provide those options, and you can provide either one without the other.

**Comment by thelinuxfoundation** [comment]
Thank you for your pull request. Before we can look at your contribution, we need to ensure all contributors are covered by a Contributor License Agreement.

After the following items are addressed, please respond with a new comment here, and the automated system will re-verify.

* One or more commits are not associated with a GitHub user. The email used in the Git commit must match a verified email address in a GitHub user profile. For more information, please see https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/

_Regards,
CLA GitHub bot_

**Comment by oneslash** [comment]
@murgatroid99, Thank you for reviewing the PR, I have fixed comments.

**Comment by murgatroid99** [comment]
Can you also add the descriptions for the purposes of the options? In particular, the fact that the first one applies to sending messages and the second applies to receiving messages.

**Comment by oneslash** [comment]
@murgatroid99 there is not much information in the header file, so I have added the proposed edits.

**Pull request merged**